### PR TITLE
feat(airdrop): Add put/get ethereum amount

### PR DESCRIPTION
**Pull request by EnzoPlayer0ne**

### CODE REVIEW

File: src/airdrop/src/lib.rs [hunk 1]
```diff
@@ -334,5 +331,70 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
     Ok(())
 }
 
+
+/// Returns all the eth addresses with how much is meant to be sent to each one of them
+#[update(guard = "caller_is_admin")]
```
Comment (chmllr) [inline src/airdrop/src/lib.rs:336]: Is this necessary?

File: src/airdrop/src/lib.rs [hunk 2]
```diff
@@ -334,5 +331,70 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
...
+pub fn get_airdrop(mut index: Index) -> CustomResult<(Index, Vec<EthAddressAmount>)> {
+    check_if_killed()?;
+
+    read_state(|state| {
+        let mut airdrop = Vec::default();
+
+        // TODO: dangerous casting from u64 to usize?
+        for idx in (index.0 as usize)..state.airdrop_reward.len() {
+            if !state.airdrop_reward[idx].transferred {
```
Comment (chmllr) [inline src/airdrop/src/lib.rs:345]: How do you know if a reward was transferred actually?

File: src/airdrop/src/lib.rs [hunk 3]
```diff
@@ -334,5 +336,67 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
     Ok(())
 }
 
+/// Returns all the eth addresses with how much is meant to be sent to each one of them
+#[update(guard = "caller_is_admin")]
+pub fn get_airdrop(mut index: Index) -> CustomResult<(Index, Vec<EthAddressAmount>)> {
```
Comment (peterpeterparker) [inline src/airdrop/src/lib.rs:341]: this seems like a bad pattern to mark this param as mutable

File: src/airdrop/src/lib.rs [hunk 4]
```diff
@@ -334,5 +336,67 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
...
+/// For a given principal check if the original airdrop amount has been redeemed
+#[query]
+pub fn has_redeemed() -> CustomResult<bool> {
+    check_if_killed()?;
+
+    let caller_principal = caller();
+
+    read_state(|state| {
+        // get the code and eth_address associated with the principal
+        let (_, eth_address) = state
```
Comment (peterpeterparker) [inline src/airdrop/src/lib.rs:385]: this duplicates the code on line 295, worth a utility?

File: src/airdrop/src/lib.rs [hunk 5]
```diff
@@ -334,5 +336,67 @@ fn bring_caninster_back_to_life() -> CustomResult<()> {
...
+            }
+        }
+    });
+
+    Ok(())
+}
+
+/// For a given principal check if the original airdrop amount has been redeemed
+#[query]
+pub fn has_redeemed() -> CustomResult<bool> {
```
[the remark this answers is not in the document]
Comment (EnzoPlayer0ne) [reply src/airdrop/src/lib.rs:381]: Redeemed == code has been used by someone. Transferred -- airdrop amount has been transferred to the Oisy Eth wallet.

File: src/airdrop/src/lib.rs (hunk 4 above)
Comment (EnzoPlayer0ne) [reply src/airdrop/src/lib.rs:388]: No, not if we integrate it with get_code as we talked about previously!

File: src/airdrop/src/lib.rs (hunk 3 above)
Comment (EnzoPlayer0ne) [reply src/airdrop/src/lib.rs:341]: I usually agree - however, I believe it makes sense as we give it an index and it returns a list of addresses along with the last index it stopped at, hence having one variable tracking the index might be easier for code comprehension.

File: src/airdrop/src/lib.rs (hunk 4 above)
Comment (peterpeterparker) [reply src/airdrop/src/lib.rs:388]: Cool then

File: src/airdrop/src/lib.rs (hunk 3 above)
Comment (peterpeterparker) [reply src/airdrop/src/lib.rs:341]: >  one variable tracking the index might be easier for code comprehension.

Well my comment proove that no. Of course it's an opiniated coding answer but, I personally would never set a params from an exposed canister update call as mutable.

Of course ultimately your call.

**Pull request merged**